src/python/dart/service/pending_actions_check.py

```diff
@@ -0,0 +1,83 @@
+from dart.context.locator import injectable
+from dart.model.workflow import WorkflowState, WorkflowInstanceState
+from dart.model.action import ActionState
+
+import logging
+import boto3
+
+_logger = logging.getLogger(__name__)
+
+@injectable
+class PendingActionsCheck(object):
+    def __init__(self, action_service):
+        self._action_service = action_service
+        self._batch_client = boto3.client('batch')
+
+    def get_not_completed_workflow_instances(self, workflow_id, workflow_service):
+        wf = workflow_service.get_workflow(workflow_id, raise_when_missing=False)
+        if not wf:
+            _logger.info('Zombie Check: workflow (id={wf_id}) not found. log-info: {log_info}'.
+                         format(wf_id=workflow_id, log_info=workflow_msg.get('log_info')))
+            return None
+
+        if wf.data.state != WorkflowState.ACTIVE:
+            _logger.info('Zombie Check: expected workflow (id={wf_id}) to be in ACTIVE state. log-info: {log_info}'.
+                         format(wf_id=workflow_id, log_info=workflow_msg.get('log_info')))
+
+        # get all workflow_instances of current workflow:
+        NOT_COMPLETE_STATES = ['QUEUED', 'RUNNING']
+        all_wf_instances = workflow_service.find_workflow_instances(workflow_id)
```

Comment:
this could use find_workflow_instances_query to avoid pulling back every workflow_instance back
it accepts a "states" param

Reply:
If I use find_workflow_instances_query I will need to use .to_model() which seems less readable,  I think it makes sense to add states param to find_workflow_instances method.

Reply:
agreed